### PR TITLE
[#3186] resources-sorting(connects #3186)

### PR DESCRIPTION
**Pull request by yuviii**

### CODE REVIEW

**Comment by yuviii** [comment]
@paulbert tried to sort resources with the `createdDate` in descending order by passing the optional arguments in the `getAll` function, but still not succeed. Need help.

**Comment by paulbert** [comment]
The issue might be that there is no index for that field on resources.  Adding the below line to `couchdb-setup.sh` and running the script would fix that
```
upsert_doc resources _index '{"index":{"fields":[{"createdDate":"desc"}]},"name":"time-index"}' POST
```

File: src/app/resources/resources.service.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class ResourcesService {
 
   requestResourcesUpdate(parent: boolean, fetchRating: boolean = true) {
     this.isActiveResourceFetch = true;
-    this.stateService.requestData(this.dbName, parent ? 'parent' : 'local');
+    this.stateService.requestData(this.dbName, parent ? 'parent' : 'local',  0, [ { 'createdDate': 'desc' } ]);
```
Comment (paulbert) [inline src/app/resources/resources.service.ts]: extra space

**Comment by yuviii** [comment]
@paulbert After indexing the `createdDate` field in resource it worked now. Please verify the change. Thank you.

**Pull request merged**